fix(tokens): DLT-1961 output font related tokens as rem instead of px

braddialpad opened the pull request:
# fix(tokens): DLT-1961 output font related tokens as rem instead of px

<!--- Feel free to remove any unused sections -->

## Obligatory GIF (super important!)

<!--- go to giphy.com, pick a gif, click share, copy gif link, paste within the () brackets below. -->

![Obligatory GIF](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExamZpejF2a2xuNW9vczMxNXZqMXl2NXltZDN0bDM4ZWY0NmY3ejB4MSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/75P7GlQ7djBDy/giphy.gif)

## :hammer_and_wrench: Type Of Change

<!--- Tick or place an `x` for the type of change. Should match the type in the commit message / PR title
in https://github.com/dialpad/dialtone/blob/staging/.github/COMMIT_CONVENTION.md -->

These types will increment the version number on release:

- [x] Fix

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1961

## :book: Description

add a new line to postcss to convert px to rem

## :bulb: Context

Our base font size is set in px (10px) so when we switched to using reference tokens and added calc() all values referencing base font size would remain in px instead of rem. This caused the issue mentioned in the jira ticket. Just added some postcss to convert the root font size to use 1rem instead of 10px when referencing base font size.

## :pencil: Checklist

<!--- Tick or place an `x` in all of the checkboxes that apply -->
<!--- Remove checkboxes that do not apply -->

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

## :crystal_ball: Next Steps

Get UC team to confirm it is fixed.

## :camera: Screenshots / GIFs

![Screenshot 2024-11-29 at 3 55 12 PM](https://github.com/user-attachments/assets/bfd468fa-bdc7-4b66-b5a0-a0463fa4da3e)

braddialpad reviewed (approved):
Looks good, just a question about the build process.

braddialpad commented:
Something seems really messed up with the package-lock. I deleted and reinstalled. Can you pull that down and do an npm install and make sure everything still works for you locally?

juliodialpad commented on docs/.vuepress/clientAppEnhance.js:
```diff
@@ -21,9 +21,12 @@ import IconWarning from "@svgIcons/IconWarning.vue";
+import IconPhone from "@svgIcons/IconPhone.vue";
+
+// Import compiled dialtone to include responsive variations
+import '../../lib/dist/css/dialtone.min.css';
```
Yes, due to the .min.css, I'll change it to use the none minified version, but we are going to change the gulp script anyway, this is just a temporary solution.

braddialpad replied:
alright, fine for now, we can merge this and finish off configuration changes in the future. We can talk about this at grooming tomorrow.

juliodialpad commented:
> Something seems really messed up with the package-lock. It kept freezing on build unless I deleted and rebuilt it. pushed this new one to this branch. Can you pull that down and do an npm install and make sure everything still works for you locally?

It's messed up, when I updated my package.json with that one, I got a warning

`warning an anonymous plugin or theme was detected and ignored`

And it doesn't detects the theme at all, I'm checking it to fix it.

braddialpad commented:
> > Something seems really messed up with the package-lock. It kept freezing on build unless I deleted and rebuilt it. pushed this new one to this branch. Can you pull that down and do an npm install and make sure everything still works for you locally?
> 
> It's messed up, when I updated my package.json with that one, I got a warning
> 
> `warning an anonymous plugin or theme was detected and ignored`
> 
> And it doesn't detects the theme at all, I'm checking it to fix it.

hmm wonder if there was a breaking change in a newer version of vuepress

juliodialpad commented:
> > > Something seems really messed up with the package-lock. It kept freezing on build unless I deleted and rebuilt it. pushed this new one to this branch. Can you pull that down and do an npm install and make sure everything still works for you locally?
> > 
> > 
> > It's messed up, when I updated my package.json with that one, I got a warning
> > `warning an anonymous plugin or theme was detected and ignored`
> > And it doesn't detects the theme at all, I'm checking it to fix it.
> 
> hmm wonder if there was a breaking change in a newer version of vuepress

I think so, the `config.ts` has some errors now

braddialpad committed:
fix(tokens): output font related tokens as rem instead of px

ninamarina reviewed (approved):
LGTM

Previously:
<img width="427" alt="image" src="https://github.com/user-attachments/assets/18e1dccf-c317-468e-ae0a-d2644421af3b">

Now:
<img width="290" alt="image" src="https://github.com/user-attachments/assets/2654653b-a8b9-4a6c-91ac-ed646ca68e96">

francisrupert reviewed (approved):
Yes, much better.
